lib/rules/constructor-super.js

```diff
@@ -31,6 +25,57 @@ function isConstructorFunction(node) {
     );
 }
 
+/**
+ * Checks whether a given node can be a constructor or not.
+ *
+ * @param {ASTNode} node - A node to check.
+ * @returns {boolean} `true` if the node can be a constructor.
+ */
+function isPossibleConstructor(node) {
+    if (!node) {
+        return false;
+    }
+
+    switch (node.type) {
+        case "ClassExpression":
+        case "FunctionExpression":
+        case "ThisExpression":
+        case "MemberExpression":
+        case "CallExpression":
+        case "NewExpression":
+        case "YieldExpression":
+        case "TaggedTemplateExpression":
+        case "MetaProperty":
+            return true;
+
+        case "Identifier":
+            return node.name !== "undefined";
```

Comment:
Yeah, I thought shadowing, but I know there was a well-known pattern past:

```js
!(function(undefined) {
    //...
}());
```

So I didn't check it.